ystia/yorc
# Feature/gh 201 use ssh agent

# Pull Request description

**Use ssh-agent to not write ssh private keys on disk** 
As: An ops
I want: to ensure that connection credentials are not left non-encrypted on disk
So that: I can relax and sleep this night

AC1: Allow to either specify a key path or a key content into the credential part of a Compute
AC2: before running Ansible command store key content into an ssh-agent for a limited amount of time
AC3: check if it also works with paramiko (anyways document what works or not)
AC4: support key content for terraform provisioning
AC5: Keys content should never be store unencrypted on disk

## Description of the change

- Add **use_ssh_agent** config option set to true by default.
- Allow using private key file if **use_ssh_agent**=false
- Allow provide private key content for Terraform check connection
- Update doc.

### What I did
**Add ssh_agent API to:**
- create new ssh-agent, add/remove keys, stop...

**Update Ansible to :**
- use ssh-agent if use_ssh_agent=true
- use private key file if use_ssh_agent=false (check if private key is a valid path)

**Update Terraform to:**
- Allow providing private key content for Terraform check connection
- Use **terraform** env var to store private key content

### How to verify it
- Deploy a topology with SSH private key content set in Compute credentials (On demand resource configuration within the A4C Location configuration)
- Config Yorc with use_ssh_agent=false and deploy the same topology with private key file path.
- Test with valid and not valid cerficates. When using  a valid certificate, you can use our home made yorc key pair, or another one created by your own, but in this latter case, you should use a tool that generates the private key in DER format (BER format not suuported)

### Description for the changelog

## Applicable Issues
fixes #201
